CoreScriptsRoot/Modules/LoadingScreen3D.lua

```diff
@@ -0,0 +1,355 @@
+	BackgroundTransparency = 1;
+	Parent = loadingSurfaceGui;
+}
+
+local spinnerRotation = 0
+local spinnerImage = Util:Create'ImageLabel'
+{
+	Name = 'Spinner';
+	Size = UDim2.new(0.25,0,0.25,0);
+	Position = UDim2.new(0.5 - (0.25/2), 0, 0.45 - (0.25/2), 0);
+	Image = 'rbxasset://textures/ui/LoadingScreen/LoadingSpinner.png';
+	BackgroundTransparency = 1;
+	Parent = backgroundImage;
+}
+
+
+local loadingText = Util:Create'TextLabel'
+{
+	Name = 'LoadingText';
+	Text = 'Loading...';
+	BackgroundTransparency = 1;
+	Font = Enum.Font.SourceSans;
+	FontSize = Enum.FontSize.Size60;
+	Position = UDim2.new(0.5,0,0.2,0);
+	Parent = backgroundImage;
+}
+
+local gameNameText = Util:Create'TextLabel'
+{
+	Name = 'GameNameText';
+	-- Text = 'Name of the Game';
+	Text = '';
+	BackgroundTransparency = 1;
+	Font = Enum.Font.SourceSans;
+	FontSize = Enum.FontSize.Size60;
+	Size = UDim2.new(0.9, 0, 0.1, 0);
+	Position = UDim2.new(0.05,0,0.65,0);
+	ClipsDescendants = true;
+	Parent = backgroundImage;
+}
+
+
+local creatorTextContainer = Util:Create'Frame'
+{
+	Name = 'CreatorTextContainer';
+	Size = UDim2.new(0.9, 0, 0.1, 0);
+	Position = UDim2.new(0.05,0,0.77,0);
+	BackgroundTransparency = 1;
+	ClipsDescendants = true;
+	Parent = backgroundImage;
+}
+
+local creatorTextPosition = 0
+local creatorText = Util:Create'TextLabel'
+{
+	Name = 'CreatorText';
+	-- Text = 'By AlsoTwentyCharacters';
+	Text = '';
+	BackgroundTransparency = 1;
+	Font = Enum.Font.SourceSans;
+	FontSize = Enum.FontSize.Size42;
+	Size = UDim2.new(1, 0, 1, 0);
+	Parent = creatorTextContainer;
+}
+
+
+local function ScreenDimsAtDepth(depth)
+	local camera = workspace.CurrentCamera
+	if camera then
+		local aspectRatio = camera.ViewportSize.x / camera.ViewportSize.y
+		local studHeight = 2 * depth * math.tan(math.rad(camera.FieldOfView/2))
+		local studWidth = studHeight * aspectRatio
+
+		return Vector2.new(studWidth, studHeight)
+	end
+	return Vector2.new(0,0)
+end
+
+local CleanedUp = false
+local freeze = true
+delay(2.5, function()
+	freeze = false
+end)
+local function UpdateLayout(delta)
+	local screenDims = ScreenDimsAtDepth(GUI_DISTANCE_FROM_CAMERA)
+
+	surfaceGuiAdorn.Size = Vector3.new(screenDims.x * HORIZONTAL_SCREEN_PERCENT, screenDims.y * VERTICAL_SCREEN_PERCENT, 1)
+	local camera = workspace.CurrentCamera
+	if camera then
+		surfaceGuiAdorn.Parent = camera
+	end
+
+
+	if creatorText.TextBounds.X < creatorTextContainer.AbsoluteSize.X then
+		creatorText.Position = UDim2.new(0, 0, 0, 0)
+		creatorText.Size = UDim2.new(1, 0, 1, 0)
+	elseif delta ~= nil then
+		creatorText.Size = UDim2.new(0, creatorText.TextBounds.X, 1, 0)
+		if not freeze then
+			local newX = (creatorTextPosition - delta * TEXT_SCROLL_SPEED)
+			if newX + creatorText.AbsoluteSize.X < creatorTextContainer.AbsoluteSize.X then
+				freeze = true
+				spawn(function()
+					Util:TweenProperty(creatorText, 'TextTransparency', creatorText.TextTransparency, 1, 1, Util:GetEaseInOutQuad())
+					wait(1.5)
```

Comment:
not enough coffee apparently

On Fri, Apr 1, 2016 at 3:12 PM, Kyle Emmerich <notifications@github.com>
wrote:

> In CoreScriptsRoot/Modules/LoadingScreen3D.lua
> <https://github.com/ROBLOX/Core-Scripts/pull/359#discussion_r58275048>:
>
> > +		surfaceGuiAdorn.Parent = camera
> > +	end
> > +
> > +
> > +	if creatorText.TextBounds.X < creatorTextContainer.AbsoluteSize.X then
> > +		creatorText.Position = UDim2.new(0, 0, 0, 0)
> > +		creatorText.Size = UDim2.new(1, 0, 1, 0)
> > +	elseif delta ~= nil then
> > +		creatorText.Size = UDim2.new(0, creatorText.TextBounds.X, 1, 0)
> > +		if not freeze then
> > +			local newX = (creatorTextPosition - delta * TEXT_SCROLL_SPEED)
> > +			if newX + creatorText.AbsoluteSize.X < creatorTextContainer.AbsoluteSize.X then
> > +				freeze = true
> > +				spawn(function()
> > +					Util:TweenProperty(creatorText, 'TextTransparency', creatorText.TextTransparency, 1, 1, Util:GetEaseInOutQuad())
> > +					wait(1.5)
>
> I'm surprised you caught that it was in RenderStepped but not that it was
> within a spawn()'d function
>
> —
> You are receiving this because you commented.
> Reply to this email directly or view it on GitHub
> <https://github.com/ROBLOX/Core-Scripts/pull/359/files/c4e35034507b4aba6fa04a83bcfc45892edf1617#r58275048>
>



-- 
ben